src/EA.Weee.Web.Tests.Unit/Areas/Producer/Controllers/ProducerControllerTests.cs

```diff
@@ -396,14 +406,48 @@ public async Task OrganisationDetails_SetViewBreadcrumb()
         }
 
         [Fact]
-        public void OrganisationDetails_Get_ShouldHaveSmallProducerSubmissionContextAttribute()
+        public void ContactDetails_Get_ShouldHaveSmallProducerSubmissionContextAttribute()
+        {
+            // Arrange
+            var methodInfo = typeof(ProducerController).GetMethod("ContactDetails", new Type[0]);
+
+            // Act & Assert
+            methodInfo.Should().BeDecoratedWith<SmallProducerSubmissionContextAttribute>();
+            methodInfo.Should().BeDecoratedWith<HttpGetAttribute>();
+        }
+
+        [Fact]
+        public void ServiceOfNoticeDetails_Get_ShouldHaveSmallProducerSubmissionContextAttribute()
         {
             // Arrange
-            var methodInfo = typeof(ProducerController).GetMethod("OrganisationDetails", new Type[0]);
+            var methodInfo = typeof(ProducerController).GetMethod("ServiceOfNoticeDetails", new Type[0]);
 
             // Act & Assert
             methodInfo.Should().BeDecoratedWith<SmallProducerSubmissionContextAttribute>();
             methodInfo.Should().BeDecoratedWith<HttpGetAttribute>();
         }
+
+        [Fact]
+        public async Task ServiceOfNoticeDetails_Get_ReturnViewModelAndMapsData()
+        {
+            // Arrange
+            SetupDefaultControllerData();
+
+            var source = new SmallProducerSubmissionMapperData
+            {
```

Comment:
@stevelee-civica  Think this test should test that the mapper is called and that the ServiceOfNoticeViewModel property on the model is that value returned from the A.CallTo(.....